portfolio/src/main/java/com/google/sps/servlets/ServeCommentsServlet.java

```diff
@@ -45,19 +66,52 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
 
   @Override
   public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    String email = request.getParameter("email");
     String content = request.getParameter("content");
-    long timestamp = System.currentTimeMillis();
+    String userTokenId = extractUserTokenId(request);
+
+    try {
+      GoogleIdToken idToken = validateUserIdToken(userTokenId);
+      System.out.println("Token ID is valid.");
+      addCommentToDb(idToken, content);
+      response.sendRedirect("/leave-comment/leave-comment.html");
+    } catch (GeneralSecurityException e) {
+      System.out.println("Token ID was not valid.");
+      response.sendError(HttpServletResponse.SC_BAD_REQUEST, e.getMessage());
+    }
+  }
+
+  private static String extractUserTokenId(HttpServletRequest request) {
+    String userTokenId = request.getParameter("userTokenId");
+    if (userTokenId == null) {
+      throw new IllegalArgumentException("User token ID is empty.");
```

Comment:
Does `IllegalArgumentException` make the servlet return a 400 (or some other 4xx)? If no, perhaps inlining this method and doing this check directly in `doPost` would be better.

Reply:
`IllegalArgumentException` should make a 400, if I am not mistaken.

Reply:
Checked it, I forced my code to throw IllegalArgumentException and got this:
HTTP ERROR 500 java.lang.IllegalArgumentException